src/robothub/replay_camera.py

```diff
@@ -0,0 +1,542 @@
+        self._color_order: dai.ColorCameraProperties.ColorOrder = dai.ColorCameraProperties.ColorOrder.BGR
+        self._interleaved = False
+        self._loop = loop
+
+        self._path_type: PathType
+        self._path = self._parse_src(src)
+        if self._path is None:
+            logging.error(f"Error in parsing src: {src}")
+            return
+
+        self.INPUT_CONTROL_STREAM_NAME = "input_control"
+        self.INPUT_CONFIG_STREAM_NAME = "input_config"
+        self.RAW_STREAM_NAME = "raw"
+        self.ISP_STREAM_NAME = "isp"
+        self.VIDEO_STREAM_NAME = "video"
+        self.STILL_STREAM_NAME = "still"
+        self.PREVIEW_STREAM_NAME = "preview"
+
+        # NOTE(miha): Used for saving references to nodes if we want to set max
+        # data size later (i.e. calling setPreviewSize also alter max data size).
+        self._nodes = {}
+
+        self._input_control = self._create_cam_input(pipeline, self.INPUT_CONTROL_STREAM_NAME)
+        self._input_config = self._create_cam_input(pipeline, self.INPUT_CONFIG_STREAM_NAME)
+        self._raw = self._create_cam_output(pipeline, self.RAW_STREAM_NAME)
+        self._isp = self._create_cam_output(pipeline, self.ISP_STREAM_NAME)
+        self._video = self._create_cam_output(
+            pipeline, self.VIDEO_STREAM_NAME, self._video_width * self._video_height * 3
+        )
+        self._still = self._create_cam_output(pipeline, self.STILL_STREAM_NAME)
+        self._preview = self._create_cam_output(
+            pipeline,
+            self.PREVIEW_STREAM_NAME,
+            self._preview_width * self._preview_height * 3,
+        )
+        # self.img_manip = pipeline.create(dai.node.ImageManip)
+        # self.img_manip.initialConfig.setResizeThumbnail(self.preview_width, self.preview_height)
+        # self.img_manip.initialConfig.setFrameType(dai.ImgFrame.Type.BGR888p)
+        # self.preview = self.img_manip.out
+
+    def _parse_src(self, src: str) -> Optional[pathlib.Path]:
+        path = pathlib.Path(src).resolve()
+        if path.is_file():
+            self._path_type = PathType.VIDEO
+            return path
+        elif path.is_dir():
+            self._path_type = PathType.IMAGE_DIRECTORY
+            return path
+        else:
+            logging.error(f"provided file: {src} is not a file or not a dir")
+
+    def _create_cam_input(self, pipeline: dai.Pipeline, stream_name: str):
+        node = pipeline.createXLinkOut()
+        node.setStreamName(stream_name)
+        self._nodes[stream_name] = node
+        return node.input
+
+    def _create_cam_output(self, pipeline: dai.Pipeline, stream_name: str, max_data_size: int | None = None):
+        node = pipeline.create(dai.node.XLinkIn)
+        node.setStreamName(stream_name)
+        if max_data_size is not None:
+            node.setMaxDataSize(max_data_size)
+        self._nodes[stream_name] = node
+        return node.out
+
+    def send_video_frames(self, device: dai.Device):
```

Comment:
why is this a public method? By the looks of it the expected way to start the replay node is `start_polling`